### PR TITLE
Geo/remove old K0 materials

### LINKED ISSUE #12966: [GeoMechanicsApplication] Remove old K0 when other is accepted
Removing the plane strain and 3D k0 material.
Add a comment in the code that this K0 procedure will be removed.

### PR DESCRIPTION
**Pull request by WPK4FEM**
**📝 Description**
The K0 procedure process handles derivation of horizontal from vertical stresses. Stresses can be a result of any constitutive law or stress initialization in another way. Therefore the geo constitutive laws with K0 in their description are now redundant.

**📝 Changes**
- Replaced use of constitutive laws with K0 in their name in all integration tests.
- Removed registry of these constitutive laws
- Removed implementations of these constitutive laws

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/test_k0_procedure_process/test_k0_procedure_k0_nc/MaterialParameters1.json [hunk 1]
```diff
@@ -23,16 +23,10 @@
           "POISSON_RATIO": 0.2,
           "THICKNESS": 1.0,
           "K0_MAIN_DIRECTION": 1,
-          "K0_VALUE_XX": 0.5,
-          "K0_VALUE_YY": 0.5,
-          "K0_VALUE_ZZ": 0.5,
-	  "K0_NC": 0.6,
+          "K0_NC": 0.6,
           "RETENTION_LAW": "SaturatedLaw",
           "RESIDUAL_SATURATION": 0.06203,
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_k0_procedure_process/test_k0_procedure_k0_nc/MaterialParameters1.json]: If I'm not mistaken, `"SaturatedLaw"` doesn't use `"RESIDUAL_SATURATION"`. If that is correct, can we remove that input parameter here?

**Review by avdg81 — COMMENTED** [review]
I really like this clean up action. Thanks for taking out the old K0 constitutive laws that we no longer want to support. This helps to make our code base simpler and thus easier to understand. I have only one very minor suggestion.

File: applications/GeoMechanicsApplication/tests/test_k0_procedure_process/test_k0_procedure_k0_nc/MaterialParameters1.json (hunk 1 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_k0_procedure_process/test_k0_procedure_k0_nc/MaterialParameters1.json]: Also removed the redundant MINIMUM_RELATIVE_PERMEABILITY.

**Review by avdg81 — APPROVED** [review]
This PR looks ready to go.

**Pull request merged**